pinot-query-runtime/src/main/java/org/apache/pinot/query/service/QueryDispatcher.java

```diff
@@ -89,7 +89,9 @@ public int submit(long requestId, QueryPlan queryPlan, long timeoutMs)
           Worker.QueryResponse response = client.submit(Worker.QueryRequest.newBuilder().setStagePlan(
                   QueryPlanSerDeUtils.serialize(constructDistributedStagePlan(queryPlan, stageId, serverInstance)))
               .putMetadata(QueryConfig.KEY_OF_BROKER_REQUEST_ID, String.valueOf(requestId))
-              .putMetadata(QueryConfig.KEY_OF_BROKER_REQUEST_TIMEOUT_MS, String.valueOf(timeoutMs)).build());
+              .putMetadata(QueryConfig.KEY_OF_BROKER_REQUEST_TIMEOUT_MS, String.valueOf(timeoutMs))
+              .putAllMetadata(queryOptions).build());
```

Comment:
should we only filter out queryOptions and attach only those that is listed in `QueryConfig.KEY_OF_*`?

Reply:
it is easier to pass in query option because we may want to pass a lot more things in the future

Reply:
on a 2nd thought i agree we should allow passing everything. current V1 engine also doesn't limit the KEY options. we can add if we saw issues in the future